repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -86,16 +91,24 @@ class Interpreter(prompt0: Ref[String],
     @tailrec def loop(blocks: Seq[Preprocessor.Output], imports: Seq[Seq[ImportData]]): Unit = {
       if(!blocks.isEmpty){
         Timer("processScript loop 0")
+        //imports from scripts loaded from this script block will end up in this buffer
+        val nestedScriptImports = mutable.Buffer.empty[ImportData]
+        scriptImportCallback = { imports => nestedScriptImports ++= imports }
         val Preprocessor.Output(code, _) = blocks.head //pretty printing results is disabled for scripts
         Timer("processScript loop 1")
         val ev = evaluate(code, imports.flatten)
         Timer("processScript loop 2")
         ev match {
           case Res.Failure(msg) => throw new CompilationError(msg)
-          case Res.Success(ev) => loop(blocks.tail, imports :+ ev.imports)
+          case Res.Success(ev) => loop(blocks.tail, imports :+ (ev.imports ++ nestedScriptImports))
           case _ => loop(blocks.tail, imports)
         }
-      } else imports.lastOption.foreach(eval.update(_))
+      } else {
+        //if we have imports to pass to the upper layer we do that
+        imports.lastOption.foreach(outerScriptImportCallback)
+        //restore the old value
+        scriptImportCallback = outerScriptImportCallback
```

Comment:
Could we put this in a `finally` block? We want it to properly reset in the case of an unhandled exception